src/celeritas/decay/interactor/MuDecayInteractor.hh

```diff
@@ -0,0 +1,242 @@
+
+    //// HELPER FUNCTIONS ////
+
+    // Boost four vector from the rest frame to the lab frame
+    inline CELER_FUNCTION FourVector to_lab_frame(Real3 const& dir,
+                                                  MevMomentum momentum,
+                                                  Mass mass) const;
+
+    // Calculate particle momentum (or kinetic energy) in the center of mass
+    inline CELER_FUNCTION MevMomentum calc_momentum(real_type energy_frac,
+                                                    Mass mass) const;
+};
+
+//---------------------------------------------------------------------------//
+// INLINE DEFINITIONS
+//---------------------------------------------------------------------------//
+/*!
+ * Construct with shared and state data.
+ *
+ * \note Geant4 physics manual defines E_{max} = m_\mu / 2, while the source
+ * code (since v10.2.0 at least) defines E_{max} = m_\mu / 2 - m_e . The source
+ * code implementation leads to a total CM energy of ~104.6 MeV instead of the
+ * expected 105.7 MeV (muon mass), which is achieved by using the physics
+ * manual definition.
+ */
+CELER_FUNCTION
+MuDecayInteractor::MuDecayInteractor(MuDecayData const& shared,
+                                     ParticleTrackView const& particle,
+                                     Real3 const& inc_direction,
+                                     StackAllocator<Secondary>& allocate)
+    : shared_(shared)
+    , inc_energy_(particle.energy())
+    , inc_direction_(inc_direction)
+    , allocate_(allocate)
+    , is_muon_(particle.particle_id() == shared_.ids.mu_minus)
+    , inc_fourvec_({inc_direction_ * particle.momentum().value(),
+                    particle.total_energy().value()})
+    , sample_electron_nu_(0,
+                          real_type{1}
+                              + ipow<2>(shared_.electron_mass.value()
+                                        / shared_.muon_mass.value()))
+    , max_energy_(real_type{0.5} * shared_.muon_mass.value()
+                  - shared_.electron_mass.value())
+{
+    CELER_EXPECT(shared_);
+    CELER_EXPECT(particle.particle_id() == shared_.ids.mu_minus
+                 || particle.particle_id() == shared_.ids.mu_plus);
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Sample the muon decay.
+ *
+ * The decay is sampled at the muon rest frame (based on
+ * \c G4MuonDecayChannel::DecayIt ) and the resulting four vector is boosted to
+ * the lab frame using the muon's incident energy and direction.
+ *
+ * Since only the charged lepton is being returned, these steps are ommitted:
+ * - Calculate the angle between charged lepton and electron neutrino.
+ * - Calculate the energies of the neutrinos.
+ * - Calculate the final directions of the neutrinos using the aforementioned
+ * angle and conservation of momentum.
+ * - Sample a spherically uniform direction and rotate all three secondaries.
+ * - Boost all particles to the lab frame
```

Comment:
Move this documentation to the main class body docs, so that it shows up in the user manual.